grr/server/grr_response_server/gui/ui/components/flow_details/helpers/osquery_test_utils.ts

```diff
@@ -0,0 +1,208 @@
+import { OsqueryColumn, OsqueryRow } from '@app/lib/api/api_interfaces';
+import { newFlowListEntry, newFlowResultSet } from '@app/lib/models/model_test_util';
+import { DeepMutable } from '@app/lib/type_utils';
+import { FlowListEntry, FlowState, FlowResultSet } from '@app/lib/models/flow';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+type FullMutableOsqueryTable = {
+  query: string,
+  header: {
+    columns: ReadonlyArray<OsqueryColumn>,
+  },
+  rows: ReadonlyArray<OsqueryRow>
+};
+
+function newOsqueryTable(): FullMutableOsqueryTable {
+  return {
+    query: '',
+    header: {
+      columns: [] as ReadonlyArray<OsqueryColumn>,
+    },
+    rows: [] as ReadonlyArray<OsqueryRow>,
+  };
+}
+
+export class OsqueryTableBuilder {
```

Comment:
Conceptually using a builder concept feels nice, but the size of the builder classes is scary. It's 150 lines - comparable to the size of the test itself. Also, the level of verbosity/complexity feels somewhat javaesque to me. I think we can rely on Javascript/Typescript features to express a very similar pattern.

What you can do is define:
```
const EMPTY_OSQUERY_TABLE: OsqueryTable {
    query: '',
    header: {
      columns: [],
    },
    rows: [],
};
```

For example:
```
const tableBuilder = new OsqueryTableBuilder()
      .withColumns(columns)
      .withValues(values);
```

can be expressed as

```
{
   ...EMPTY_OSQUERY_TABLE,
   header: {
      columns,
   },
   rows: values.map((rowValues) => ({values: rowValues}))
}
```

if you introduce a single helper function `valuesToRows`, it will look even better:

```
{
   ...EMPTY_OSQUERY_TABLE,
   header: {
      columns,
   },
   rows: valuesToRows(values)
}
```